pkg/applylib/applyset/applyset.go

```diff
@@ -141,7 +144,24 @@ func (a *ApplySet) ApplyOnce(ctx context.Context) (*ApplyResults, error) {
 			return nil, fmt.Errorf("unknown scope for gvk %s: %q", gvk, restMapping.Scope.Name())
 		}
 
-		j, err := json.Marshal(obj)
+		currentObj, err := dynamicResource.Get(ctx, name, v1.GetOptions{})
```

Comment:
I do wonder whether we should put this into a separate method, because (I would hope) this isn't going to be the common path.

One possible approach .... when we apply, we get the "after" object.  We could inspect that to see if field ownership changes are needed.  It's slightly less efficient in this case, because I'd imagine we'd just reprocess every object, but I would expect this case also to be the rare case.